slo-monitor/src/monitors/store.go

```diff
@@ -0,0 +1,196 @@
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package monitors
+
+import (
+	"sync"
+
+	"k8s.io/client-go/tools/cache"
+
+	"github.com/golang/glog"
+)
+
+const (
+	updateOperationType = "update"
+	deleteOperationType = "delete"
+)
+
+type workItem struct {
+	operationType string
+	key           string
+	obj           interface{}
+}
+
+// NoStoreQueue implements k8s.io/client-go/tools/cache Store and Queue interfaces
+type NoStoreQueue struct {
+	sync.Mutex
+
+	cond      sync.Cond
+	queue     []workItem
+	indices   map[string]int64
+	nextIndex int64
+	offset    int64
+}
+
+var keyFunc = cache.DeletionHandlingMetaNamespaceKeyFunc
+
+// NewNoStoreQueue returns newly created NoStoreQueue
+func NewNoStoreQueue() *NoStoreQueue {
+	q := &NoStoreQueue{
+		queue:     []workItem{},
+		indices:   map[string]int64{},
+		nextIndex: 0,
+		offset:    0,
+	}
+
+	q.cond.L = q
+	return q
+}
+
+func (q *NoStoreQueue) addOrModifyItem(key string, obj interface{}, operationType string) {
+	if index, ok := q.indices[key]; ok {
+		q.queue[int(int64(index)-q.offset)].operationType = operationType
+		q.queue[int(int64(index)-q.offset)].obj = obj
+		return
+	}
+	q.indices[key] = q.nextIndex
+	q.nextIndex++
+	q.queue = append(q.queue, workItem{
+		operationType: operationType,
+		key:           key,
+		obj:           obj,
+	})
+}
+
+// Add implements Add function from Store interface
+func (q *NoStoreQueue) Add(obj interface{}) error {
+	key, err := keyFunc(obj)
+	if err != nil {
+		return cache.KeyError{obj, err}
+	}
+
+	q.Lock()
+	defer q.Unlock()
+	defer q.cond.Broadcast()
+	q.addOrModifyItem(key, obj, updateOperationType)
+	return nil
+}
+
+// Update implements Update function from Store interface
+func (q *NoStoreQueue) Update(obj interface{}) error {
+	return q.Add(obj)
+}
+
+// Delete implements Delete function from Store interface
+func (q *NoStoreQueue) Delete(obj interface{}) error {
+	key, err := keyFunc(obj)
+	if err != nil {
+		return cache.KeyError{obj, err}
+	}
+
+	q.Lock()
+	defer q.Unlock()
+	defer q.cond.Broadcast()
+	q.addOrModifyItem(key, obj, deleteOperationType)
+	return nil
+}
+
+// List implements List function from Store interface
+func (q *NoStoreQueue) List() []interface{} {
+	return []interface{}{}
+}
+
+// ListKeys implements ListKeys function from Store interface
+func (q *NoStoreQueue) ListKeys() []string {
+	return []string{}
+}
+
+// Get implements Get function from Store interface
+func (q *NoStoreQueue) Get(obj interface{}) (item interface{}, exists bool, err error) {
+	return "", false, nil
```

Comment:
Let's return "unimplemented" errors in all methods - should be easier to discover if someone will start using it by mistake.